TLM/TLM/Util/Shortcuts.cs

```diff
@@ -19,17 +17,35 @@ internal static class Shortcuts {
         internal static IList<T> Clone<T>(this IList<T> listToClone) where T : ICloneable =>
             listToClone.Select(item => (T)item.Clone()).ToList();
 
+        internal static void Swap<T>(ref T a, ref T b) {
```

Comment:
C# 7 introduced tuples which enables swapping two variables without a temporary one:
```
int a = 10;
int b = 2;
(a, b) = (b, a);
```

Reply:
Yeah I know! but this is not c# 7 :(

Reply:
We have all C# versions available. Only .NET framework is limited to 3.5.

Reply:
Actually... not. Some of the features still need additional namespaces not available for .net 3.5, sadly :/

Reply:
@kvakvs Well, in fact, there will be even **two** temporary variables with your solution. Check the generated IL. And @krzychu124 is right: the .NET Framework 3.5 BCL has no value tuples, as well as Mono of Unity v5, so this is not possible without an additional NuGet package (which I wouldn't recommend).